### ISSUE TITLE
[sim-services] Make I3PropagatorModule propagate only particles with 0/NaN length? (Trac #2128)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2120]
This might not be relevant for most people, but I have an application where I run I3Propagator on an MCTree, where some particles have already been propagated. Since I do not want to propagate them twice, an obvious fix would be to check the length of the particles and only propagate them, if their length is 0 or NaN.

Would this change break anything?

Migrated from https://code.icecube.wisc.edu/ticket/2128
```json
{
    "status": "closed", 
    "changetime": "2019-03-19T13:13:02", 
    "description": "This might not be relevant for most people, but I have an application where I run I3Propagator on an MCTree, where some particles have already been propagated. Since I do not want to propagate them twice, an obvious fix would be to check the length of the particles and only propagate them, if their length is 0 or NaN.\n\nWould this change break anything?", 
    "reporter": "chaack", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1553001182278345", 
    "component": "combo simulation", 
    "summary": "[sim-services] Make I3PropagatorModule propagate only particles with 0/NaN length?", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2018-01-12T09:43:23", 
    "milestone": "Vernal Equinox 2019", 
    "owner": "olivas", 
    "type": "enhancement"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20180922T09:09:20: olivas commented:

> This is the way it worked at some point.  I'll look into this.

**Comment by icecube-trac** [comment]
Trac update at 20180922T09:09:33: 
* olivas changed milestone from "" to "Winter Solstice 2018"
* olivas changed status from "new" to "assigned"
* olivas changed owner from "" to "olivas"

**Comment by icecube-trac** [comment]
Trac update at 20181114T14:39:03: olivas changed owner from "olivas" to ""

**Comment by icecube-trac** [comment]
Trac update at 20181127T02:36:27: olivas changed owner from "" to "jsoedingrekso"

**Comment by icecube-trac** [comment]
Trac update at 20190318T20:40:11: olivas commented:

> Slight tweak to the ticket.  I3Propagator is an abstract base class, where propagate is a pure virtual method, so this'll have to be implemented in derived classes.  I was able to find 3 derived classes, one each for for PROPOSAL, cmc, and nugen.  I'm guessing it's fine if this is implemented in just I3PropagatorServicePROPOSAL.

**Comment by icecube-trac** [comment]
Trac update at 20190319T12:14:20: 
* olivas commented:

> This change should be made in the module and not the propagator service.


* olivas changed summary from "Make I3Propagator propagate only particles with 0/NaN length?" to "[sim-services] Make I3PropagatorModule propagate only particles with 0/NaN length?"
* olivas changed status from "assigned" to "accepted"
* olivas changed owner from "jsoedingrekso" to "olivas"

**Comment by icecube-trac** [comment]
Trac update at 20190319T13:13:02: 
* olivas changed resolution from "" to "fixed"
* olivas commented:

> In changeset 172144:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="172144"
> add the option to skip NaN length nodes in the propagation.  closes #2120
> ```


* olivas changed status from "accepted" to "closed"
